File: js/leetcode/move-zeroes.ts
// move 0,2,0,12 to 2,12,00

var moveZeroes = function (nums: number[]) {
  let pointer = 0;
  const length = nums.length;
  for (let i = 0; i < length; i++) {
    if (nums[i] !== 0) {
      nums[pointer] = nums[i];
      pointer++;
    }
  }

  for (let i = pointer; i < length; i++) {
    nums[i] = 0;
  }
  return nums;
};

File: go/leetcode/arrays/move_zeroes.go
package leetcode

func MoveZeroes(nums []int) {
	pointer := 0
	i := 0

	for i = range nums {

		if nums[i] != 0 {
			nums[pointer] = nums[i]
			pointer++
		}
	}

	for i = pointer; i < len(nums); i++ {
		nums[i] = 0
	}
}
